src/Microsoft.SqlTools.ServiceLayer/InsightsGenerator/InsightsGeneratorService.cs

```diff
@@ -45,6 +46,18 @@ public void InitializeService(ServiceHost serviceHost)
 
         internal async Task HandleQueryInsightGeneratorRequest(QueryInsightsGeneratorParams parameters, RequestContext<InsightsGeneratorResult> requestContext)
         {
+            QueryExecutionService service = QueryExecutionService.Instance;
+
+            QueryExecution.Contracts.SubsetParams subsetParams = new QueryExecution.Contracts.SubsetParams{
+                OwnerUri = parameters.OwnerUri
+            }; 
+
+            QueryExecution.Contracts.ResultSetSubset result = await service.InterServiceResultSubset(subsetParams);
+            
+            Microsoft.InsightsGenerator.Workflow insightWorkFlowInstance = Microsoft.InsightsGenerator.Workflow.Instance();
+
+            insightWorkFlowInstance.IngestRules(result.Rows);
```

Comment:
do we need a result.Rows.to_array() or does it come in the DataArray format already. Appologize haven't used ti recently so I don't remember how it looks.

Reply:
I can probably send a DataArray to the IngestRules

Reply:
This PR uses dataarray : https://github.com/microsoft/sqltoolsservice/pull/1019